### DISCUSSION TITLE
Wasm - "unhandled promise rejection" after deployment

### DISCUSSION THREAD

**Discussion by garvanwalshe** [leptos-rs/leptos#2931] [Q&A] [answered]
After deploying my app to production (using a containerfile as recommended in the docs) there is no interactivity. The javascript console in Chrome reports

```Uncaught (in promise) LinkError: WebAssembly.instantiate(): Import #186 "wbg" "__wbindgen_closure_wrapper1261": function import requires callable```

Here is the relevant section in the project index file.  (The project is called leptos-dash)
```
<script type="module">
            function idle(c) {
                if ("requestIdleCallback"in window) {
                    window.requestIdleCallback(c);
                } else {
                    c();
                }
            }
            idle( () => {
                import('/pkg/leptos-dash.js').then(mod => {
                    mod.default('/pkg/leptos-dash.wasm').then( () => mod.hydrate());
                }
                )
            }
            );
        </script>
```

The code generated in the version run locally that works seems  to me to be identical 

```
 <script type="module">
            function idle(c) {
                if ("requestIdleCallback"in window) {
                    window.requestIdleCallback(c);
                } else {
                    c();
                }
            }
            idle( () => {
                import('/pkg/leptos-dash.js').then(mod => {
                    mod.default('/pkg/leptos-dash.wasm').then( () => mod.hydrate());
                }
                )
            }
            );
        </script>
```

Safari reports the same error.

Everything works fine on  my development machine (in both development and release modes) but I have had this error appear and disappear with a clean build. However, my container file builds things from scratch.   Any hints on how to go about tracking down where this error comes from?

I'm using Axum as the underlying framework.

**Comment by gbj** [comment, marked as answer]
I would say that this kind of error is typically a caching issue, related to a version of the JS file and/or the WASM file being served that is not the latest.
